src/java/fr/paris/lutece/portal/service/securityheader/SecurityHeaderService.java

```diff
@@ -0,0 +1,300 @@
+	//It must be refreshed each time that an action is performed. Currently, These actions are creating a new header, modifying an 
+	//existing header, removing a header and enabling or disabling a header
+	private Map<String, SecurityHeader> _mapSecurityHeaders = new HashMap<String, SecurityHeader>();
+	
+	//This boolean indicates if a refresh has been done on the security headers map. In other words, it tells if the map is up to date
+	//with data from database
+	private boolean _bMapRefreshDone = false;
+	
+	// Properties
+	private static final String PROPERTY_LABEL_SECURITY_HEADER_PAGE_TYPE = "portal.securityheader.labelPageType";
+	private static final String PROPERTY_LABEL_SECURITY_HEADER_REST_API_TYPE = "portal.securityheader.labelRestApiType";	
+	private static final String PROPERTY_LABEL_SECURITY_HEADER_PAGE_CATEGORY_ALL = "portal.securityheader.labelAllPageCategory";
+	private static final String PROPERTY_LABEL_SECURITY_HEADER_PAGE_CATEGORY_LOGOUT_BO = "portal.securityheader.labelLogoutBOPageCategory";
+	private static final String PROPERTY_LABEL_SECURITY_HEADER_PAGE_CATEGORY_LOGOUT_FO = "portal.securityheader.labelLogoutFOPageCategory";
+	private static final String PROPERTY_LABEL_SECURITY_HEADER_PAGE_CATEGORY_ADMIN_BO = "portal.securityheader.labelAdminBOCategory";
+	private static final String PROPERTY_LABEL_SECURITY_HEADER_PAGE_CATEGORY_ADMIN_FO =  "portal.securityheader.labelAdminFOCategory";
+	
+	private Logger _logger = LogManager.getLogger( "lutece.securityHeader" );
+	
+	/**
+	 * Returns all the security headers that match specified name, type and page category (if not null for the latter).
+	 * If page category is null (REST api case), this criterion is ignored.
+	 * 
+	 * @param strName
+	 *           The name of the security header
+	 * @param strType
+	 *           The type of the security header
+	 * @param pageCategory
+	 *           The page category of the security header
+	 * @return list od security headers matching the criteria
+	 */
+	public List<SecurityHeader> find( String strName, String strType, String pageCategory )
+	{
+		ArrayList<SecurityHeader> securityHeadersResultList = new ArrayList<SecurityHeader>( );
+		
+		for( SecurityHeader securityHeader : findAll( ) )
+		{
+			if( securityHeader.getName( ).equals( strName ) && securityHeader.getType( ).equals( strType ) )
+			{
+				if( pageCategory == null || securityHeader.getPageCategory( ).equals( pageCategory ) )
+				{
+					securityHeadersResultList.add( securityHeader );
+				}				
+			}
+		}
+		
+		return securityHeadersResultList;
+	}
+	
+	/**
+	 * Returns all security headers from the specified type and the specified category (if not null)
+	 * that are active (attribute is_enable = true).
+	 * 
+	 * @param strType
+	 * @param strPageCategory
+	 * @return collection on security headers
+	 */
+	public Collection<SecurityHeader> findActive( String strType, String strPageCategory )
```

Comment:
Taken into account in commit 93a56b8139c008a7cc3a87cce5a8ccbffb4c9734

Map type used is Map<String, Map<String, List<SecurityHeader>>> instead of Map<String, Map<String, SecurityHeader>>